exporters/nova.go

```diff
@@ -129,30 +131,48 @@ func ListHypervisors(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metri
 		return
 	}
 
+	allPagesAggregates, err := aggregates.List(exporter.Client).AllPages()
+	if err != nil {
+		log.Errorln(err)
+		return
+	}
+
+	if allAggregates, err = aggregates.ExtractAggregates(allPagesAggregates); err != nil {
+		log.Errorln(err)
+		return
+	}
+
+	hostToAggregateMap := map[string]string{}
+	for _, v := range allAggregates {
+		for _, w := range v.Hosts {
+			hostToAggregateMap[w] = v.AvailabilityZone
+		}
+	}
+
```

Comment:
I think it would be better to generate a metric per aggregate or Join the list of aggregates by comma